projectile.el

```diff
@@ -1817,36 +1821,38 @@ project-root for every file."
   (let ((prompt (projectile-prepend-project-name prompt))
         res)
     (setq res
-          (cond
-           ((eq projectile-completion-system 'ido)
-            (ido-completing-read prompt choices nil nil initial-input))
-           ((eq projectile-completion-system 'default)
-            (completing-read prompt choices nil nil initial-input))
-           ((eq projectile-completion-system 'helm)
-            (if (and (fboundp 'helm)
-                     (fboundp 'helm-make-source))
-                (helm :sources
-                      (helm-make-source "Projectile" 'helm-source-sync
-                                        :candidates choices
-                                        :action (if action
-                                                    (prog1 action
-                                                      (setq action nil))
-                                                  #'identity))
-                      :prompt prompt
-                      :input initial-input
-                      :buffer "*helm-projectile*")
-              (user-error "Please install helm from \
-https://github.com/emacs-helm/helm")))
-           ((eq projectile-completion-system 'ivy)
-            (if (fboundp 'ivy-read)
-                (ivy-read prompt choices
-                          :initial-input initial-input
-                          :action (prog1 action
-                                    (setq action nil))
-                          :caller 'projectile-completing-read)
-              (user-error "Please install ivy from \
-https://github.com/abo-abo/swiper")))
-           (t (funcall projectile-completion-system prompt choices))))
+          (pcase (if (eq projectile-completion-system 'auto)
+                     (cond
+                      ((and (boundp 'ido-mode)  ido-mode)  'ido)
```

Comment:
You can use `bound-and-true-p` for these btw.

Reply:
Thank you! Good to know! Maybe @bbatsov can fix this quickly or should I do another PR?

Reply:
It's not a big deal, just wanted to mention it.

Reply:
Yes, it's good to know. These things are easy to miss if you are unexperienced with elisp. The only thing which helps is going over the list of all functions or maybe there is some kind of linter which also checks things like this (similar to hlint).
But I must also say that the emacs code organization seems a bit chaotic. bound-and-true-p is defined in bindings.el, which does not make much sense, except that the function is just used there immediately after the definition? I went over the subr and subr-x libraries for example and would have noticed such a function...

Reply:
I'll update that myself. I keep forgetting about that function.

Reply:
@bbatsov Thank you!